internal/generator/vector/metrics.go

```diff
@@ -1,13 +1,28 @@
 package vector
 
+import (
+	"golang.org/x/sys/unix"
+)
+
 const (
 	InternalMetricsSourceName = "internal_metrics"
 	PrometheusOutputSinkName  = "prometheus_output"
-	PrometheusExporterAddress = "[::]:24231"
 
 	AddNodenameToMetricTransformName = "add_nodename_to_metric"
 )
 
+var PrometheusExporterAddress string
+
+func init() {
+	if fd, err := unix.Socket(unix.AF_INET6, unix.SOCK_STREAM, unix.IPPROTO_IP); err != nil {
+		PrometheusExporterAddress = `0.0.0.0`
```

Comment:
Have you tried using an empty hostname `":12345" rather than an IP literal? The empty hostname is supposed to listen on all local addresses for ipv6 or ipv4.

Reply:
Yep ^^^ https://github.com/openshift/cluster-logging-operator/pull/2200#discussion_r1350382865

Reply:
Argh. I have long hated getaddrinfo in C, and now I hate the equivalent in Rust as well. Is a simple "listen to everything" so much to ask?
Have you seen https://issues.redhat.com/browse/LOG-4608 ? I think we have to solve and test both together. QE may have test clusters now, and I guess we can force the ipv settings on a CRC or SNO cluster to test all scenarios.